Dockerfile

```diff
@@ -261,6 +261,23 @@ RUN curl -Os https://dl.google.com/android/repository/commandlinetools-linux-${A
     && chmod -R o+rw $ANDROID_HOME
 COPY docker/android.sh /etc/profile.d
 
+#------------------------------------------------------------------------
```

Comment:
From the commit message, I was expecting that the old way of setting up Dart (via APT) was replaced with the new way: separate layer + copy command. I believe the use of terms "move"  and "switch" lead me into thinking so.

Looking at the diff, it seems that the old way still remains and the new way is added but not used.
I believe that this whole "move" step actually is atomic, and should not be split into two commits.
If you agree with that this can be easily fixed by squashing the commit "docker: Replace Dart installation with new binary layer " into this one. (The third commit IMO is nice to have separately).

Reply:
In other PRs it was agreed with other reviewers to have these separate commits, so keep as-is as it is analog.